### DISCUSSION TITLE
Read formData() in both clientAction and action

### DISCUSSION THREAD

**Discussion by mattoni** [remix-run/react-router#12587] [Q&A]
I'm trying to follow this guide by @sergiodxa:

https://sergiodxa.com/tutorials/validate-form-in-remix-with-clientaction

However, when trying to implement both a `clientAction` and server `action` that reads `formData`, I get this error:

`Request.text: Body has already been consumed.`

Is there a way to allow `formData` to be read in both without causing this issue?

**Comment by mattoni** [comment]
I suppose i could just use request.clone() in the clientAction. Didn't realize that existed.

**Reply by frontendphil** [reply]
Thank you! I also didn't realize this was a thing. I got stuck for a day on this. It would be great to have some other mechanism in place, though. My use case is that there are certain submissions that I can (and must) handle on the client and then hand off to the server for everything else. To achieve that, I'm sending along an `intent` parameter in the `formData`. 

I forgot that I'm working on open source for a change, so I can actually share a link to my specific use case: https://github.com/gnosisguild/zodiac-pilot/blob/main/deployables/app/app/routes/edit-route.%24data.tsx
